merkle/map_hasher.go

```diff
@@ -0,0 +1,41 @@
+package merkle
+
+import (
+	"github.com/google/trillian"
+)
+
+// MapHasher is a specialised TreeHasher which also knows about the set of
+// "null" hashes for the unused sections of a SparseMerkleTree.
+type MapHasher struct {
+	TreeHasher
+	keyHasher  keyHashFunc
+	nullHashes []trillian.Hash
+}
+
+// NewMapHasher creates a new MapHasher based on the passed in hash function.
+func NewMapHasher(hasher trillian.Hasher) MapHasher {
```

Comment:
Yeah, that's probably going to be required to allow configurable Tree/Map Hashers.
Done.